File: java/src/main/java/me/anviks/codewars/solutions/_7kyu/valid_parentheses/Kata.java
/*
 * https://www.codewars.com/kata/6411b91a5e71b915d237332d
 */

package me.anviks.codewars.solutions._7kyu.valid_parentheses;

import java.util.Stack;

public class Kata {
    public static boolean validParentheses(String parens) {
        Stack<String> parentheses = new Stack<>();
        for (String character : parens.split("")) {
            if (character.equals("(")) {
                parentheses.push("(");
            } else if (character.equals(")")) {
                if (parentheses.isEmpty()) {
                    return false;
                }
                parentheses.pop();
            }
        }

        return parentheses.isEmpty();
    }
}

File: kotlin/src/main/kotlin/me/anviks/codewars/solutions/_5kyu/simple_fun_number_333_incomplete_virus/Kata.kt
/*
 * https://www.codewars.com/kata/595467c63074e38ba4000063
 */

package me.anviks.codewars.solutions._5kyu.simple_fun_number_333_incomplete_virus


object Kata {
    fun incompleteVirus(s: String): Long {
        var binaryString = ""
        var largeDigitEncountered = false

        for (char in s) {
            if (!largeDigitEncountered && char < '2') {
                binaryString += char
            } else {
                largeDigitEncountered = true
                binaryString += '1'
            }
        }

        return binaryString.toLong(2)
    }
}

